### PR TITLE
Removed build warnings when using iOS 8 or 10.10 SDKs for the Date code.

### PR DESCRIPTION
**Pull request by samkrishna**
@segiddins I've removed the build warnings for the iOS 8 and OS X 10.10 SDKs.

I have code for the `RKISO8601DateFormatter` to deal with those build warnings as well, but I can't figure out how to initiate a pull request for that code, since @blakewatters original repo doesn't have the 'RK'-prefixed version of the formatter and I don't see it in any convenient location in the RK Repo.

Please advise where/how I should open up a PR for RKISO8601DateFormatter. Thanks.

### CODE REVIEW

**Comment by segiddins** [comment]
@samkrishna can you please take a look at the test failures?

File: Code/ObjectMapping/RKHTTPUtilities.m [hunk 1]
```diff
@@ -397,11 +401,22 @@ RKRequestMethod RKRequestMethodFromString(NSString *methodName)
...
-    dispatch_once(&onceToken, ^{ gmtTimeZone = CFTimeZoneCreateWithTimeIntervalFromGMT(NULL, 0.0); });
     
+#if (__IPHONE_OS_VERSION_MIN_ALLOWED >= 80000 || MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_10)
+    static NSCalendar *gregorian;
+    dispatch_once(&onceToken, ^{
+        gregorian = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
+        gregorian.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:0];
+    });
+
+    if(parsed == 1) { date = [gregorian dateFromComponents:gdate]; }
```
Comment (segiddins) [inline Code/ObjectMapping/RKHTTPUtilities.m]: `if (parsed == 1)` and the code block on the next line, please

File: Code/ObjectMapping/RKHTTPUtilities.m [hunk 2]
```diff
@@ -397,11 +401,22 @@ RKRequestMethod RKRequestMethodFromString(NSString *methodName)
...
+#else
+    static CFTimeZoneRef gmtTimeZone;
+    dispatch_once(&onceToken, ^{ gmtTimeZone = CFTimeZoneCreateWithTimeIntervalFromGMT(NULL, 0.0); });
+
     if(parsed == 1) { date = [NSDate dateWithTimeIntervalSinceReferenceDate:CFGregorianDateGetAbsoluteTime(gdate, gmtTimeZone)]; }
```
Comment (segiddins) [inline Code/ObjectMapping/RKHTTPUtilities.m]: `if (parsed == 1)` and the code block on the next line, please

**Comment by segiddins** [comment]
:+1: once my comments are addressed.

**Comment by segiddins** [comment]
@samkrishna ping?

**Comment by segiddins** [comment]
Please address the comments so this can be merged.

**Comment by honkmaster** [comment]
@segiddins This is the same as #2238. Even if the code style does not comply with personal preference of code layout, @samkrishna provided a valid bugfix. Don't refuse such a small merge just because you don't like the formatting. Just change the two lines on your own ...

**Comment by segiddins** [comment]
@honkmaster I prefer that the code in RestKit stays consistent, otherwise there's unnecessary cognitive overhead when trying to understand the code.

**Comment by honkmaster** [comment]
I understand this, however the PR is valid and not to merge it just because of personal preference is strange. A simple merge and changing the two lines on your own would have taken less time than to reply and request for change by @samkrishna here on GitHub.

**Comment by samkrishna** [comment]
@segiddins I want to respectfully point out that much of the code formatting in 'RKHTTPUtilities.m' is formatted in a way you would flag **TODAY** if you were to be dealing with those PRs now. At the time of this original PR submission, I attempted to conform to what the rest of the source file's formatting style looked like (as ugly as **IT IS**).

In any case, I've re-arranged some code inside the preprocessor checks and formatted it in **One True Brace** style with what I think is your style guidance.

Please accept this PR.

**Comment by segiddins** [comment]
Your changes look good, but it seems there are a bunch of other unrelated commits in the PR now

**Comment by samkrishna** [comment]
OK -- I'll start another PR.

**Pull request closed without merging**